src/inference/gibbs_conditional.jl

```diff
@@ -0,0 +1,168 @@
+struct GibbsConditional{space, C} <: InferenceAlgorithm 
+    conditional::C
+
+    function GibbsConditional(sym::Symbol, conditional::C) where {C}
+        return new{sym, C}(conditional)
+    end
+end
+
+getspace(::GibbsConditional{S}) where {S} = (S,)
+alg_str(::Sampler{<:GibbsConditional}) = "GibbsConditional"
+
+
+#################
+# Transition #
+#################
+
+struct GibbsConditionalTransition{T, F<:AbstractFloat}
+    θ::T
+end
+
+function GibbsConditionalTransition(spl::Sampler{<:GibbsConditional})
+    θ = tonamedtuple(spl.state.vi)
+    return GibbsConditionalTransition(θ)
+end
+
+
+mutable struct GibbsConditionalState{V<:VarInfo} <: AbstractSamplerState
+    vi::V
+end
```

Comment:
You also don't need this, you could just use a simple `SamplerState` https://github.com/TuringLang/Turing.jl/blob/8cafa3114089d21070790f2b4e09e3d23e4bcbb8/src/inference/Inference.jl#L442-L444.